BaseClasses.py

```diff
@@ -318,6 +318,42 @@ def has_paper(self):
     # consider: can_wear(mask_name), rather than just using has(item)
     # would do a check to see if you have the mask and can do human form
 
+    def stray_fairy_req(self):
+        return self.can_use('Great Fairy Mask') or not self.options('ReqGFMask')
```

Comment:
The self.options stuff is because I'm not sure of the way of getting options that were checked in the UI; like here, if the user wanted to require that you have the great fairy mask to get all the dungeon fairies, they check that box, then (I assume) the option is recorded, and this check uses that option

Reply:
Options are set in the World class, we'll see how this goes later on. I'll just let this through as is, since most is scheduled for refactoring/review later on anyway.